python/ql/lib/semmle/python/ApiGraphs.qll

```diff
@@ -694,11 +694,11 @@ module API {
           rhs = aw.getValue()
         )
         or
-        // TODO: I had expected `DataFlow::AttrWrite` to contain the attribute writes from a dict, that's how JS works.
+        // for dictionary literals, from `x` to `{ "key": x }`
         exists(PY::Dict dict, PY::KeyValuePair item |
           dict = pred.(DataFlow::ExprNode).getNode().getNode() and
           dict.getItem(_) = item and
-          lbl = Label::member(item.getKey().(PY::StrConst).getS()) and
+          lbl = Label::subscript() and
```

Comment:
I would say it depends. If we find an API that exposes some components as numerically-indexed bits of some object, then maybe it'll be relevant.

(For that matter, I don't know for certain that _string_ indices are necessary, but it feels like it would be more likely for an API to have been constructed in this way.)

At any rate, we would still have an escape hatch where we assign an API node to the index, thus permitting more fine-grained control over the exact form of the index.

Reply:
I added a convenience predicate for `getSubscript("bar")`. It is not part of the API graph, so I wrote a separate test to show how it works. Is this the right approach, or did you wan to add labels with strings on them for the known cases?

Reply:
Ah, I did not refresh and see your comment. Maybe you want a new label `getIndex` that points to the index of a subscript?

Reply:
I think we should probably have a wider conversation about this, before committing to a specific API. I think adding `getIndex` labels definitely makes sense, though I'm not entirely sure if they source node of such a label should be the `foo` bit of `foo[bar]` or the entire subscript expression. By comparison with how arguments/parameters are modelled, I'm inclined to say it should be the former.